### PR TITLE
[Naitei] ft:Sign up

### PR DESCRIPTION
**Pull request by ThaiKhacDuoc**
[Naitei] ft:Sign up

## Related Tickets
- ticket redmine

## WHAT (optional)
- Change number items `completed/total` in admin page.

## HOW
- I edit js file, inject not_vary_normal items in calculate function.

## WHY (optional)
- Because in previous version - number just depends on `normal` items. But in new version, we have `state` and `confirm_state` depends on both `normal` + `not_normal` items.

## Evidence (Screenshot or Video)
Eng:
image

Vi:
image

### CODE REVIEW

**Comment by ThaiKhacDuoc** [comment]
ready

**Comment by ThaiKhacDuoc** [comment]
ready

File: app/models/user.rb [hunk 1]
```diff
@@ -1,10 +1,30 @@
...
 
+  validates :name, presence: true,
+  length: {maximum: Settings.maximum_name_length}
+
+  validates :email, presence: true, uniqueness: true,
+     length: {maximum: Settings.maximum_email_length},
+     format: {with: Regexp.new(Settings.email_regex)}
+
+  validates :password, presence: true,
+        length: {minimum: Settings.min_password_length}
```
Comment (ghost) [inline app/models/user.rb]: em check rubocop chưa sao indent thò thụt vậy nhỉ

File: app/views/shared/_error_messages.html.erb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<% if @user.errors.any? %>
+  <div id="error_explanation">
+    <div class="alert alert-danger">
+      <%= t("sign_up.the_form_contains") %>
```
Comment (ghost) [inline app/views/shared/_error_messages.html.erb]:
```suggestion
      <%= t "sign_up.the_form_contains" %>
```

**Review by ghost — COMMENTED** [review]
update thêm ảnh evident nhé

**Comment by ThaiKhacDuoc** [comment]
ready

File: config/initializers/assets.rb [hunk 3]
```diff
@@ -3,6 +3,10 @@
 # Version of your assets, change this if you want to expire all your assets.
 Rails.application.config.assets.version = "1.0"
 
+Rails.application.config.assets.precompile += %w( custom.css )
+Rails.application.config.assets.precompile += %w( controllers/hello_controller.js )
+Rails.application.config.assets.precompile += %w( controllers/index.js )
```
Comment (ghost) [inline config/initializers/assets.rb]:
```suggestion
Rails.application.config.assets.precompile += %w( custom.css controllers/hello_controller.js controllers/index.js)
```
Thử thế này xem em

**Review by ghost — COMMENTED** [review]
merged

**Pull request merged**